src/lib/components/VideoEmbed.svelte

```diff
@@ -13,7 +19,50 @@
 		<time datetime={video.date.toISOString()}>{video.date.toLocaleDateString()}</time>
 	</div>
 	<div class="video">
-		<FrameEmbed src="https://archive.org/embed/{video.id}" title={video.title} />
+		{#if $preferredSource == VideoSource.Direct && video.source.direct}
+			<video controls src={video.source.direct} />
+		{:else if $preferredSource == VideoSource.YouTube && video.source.youtube}
+			<FrameEmbed
+				src="https://www.youtube.com/embed/{video.source.youtube}"
+				title={video.title}
+			/>
+		{:else}
+			<FrameEmbed
+				src="https://archive.org/embed/{video.source.internetarchive}"
+				title={video.title}
+			/>
+		{/if}
+		<ul class="video-controls">
+			<li>
+				<span class="link link-gear">Source</span>
+				<div class="controls">
+					<button
+						title="Use Internet Archive video source"
+						on:click={() => setSource(VideoSource.InternetArchive)}
+						class={$preferredSource === VideoSource.InternetArchive && 'current'}
```

Comment:
Sorry I missed this the first time over, but I've never seen this method of applying a conditional class. Would it not be better to do this?

`class:current={$preferredSource === VideoSource.InternetArchive}`

and for `disabled` you only need this:
`disabled={!video.source.internetarchive}`

Reply:
This is a common pattern we use at work, but I also don't know if it's any good as I tend to avoid the frontend stuff there :P

Your suggestions were (again) much nicer so I've made the change as well as disabled the hover effect when the button is disabled.

Reply:
Glad to make good suggestions! I like the hover effect change too.